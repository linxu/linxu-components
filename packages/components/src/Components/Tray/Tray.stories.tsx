import React from "react";
import { Story } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import { getTheme } from "../../theme/getTheme";
import { ExomeTray, TrayProps } from "./Tray";

export default {
  title: "Components/Tray",
  component: ExomeTray,
};

export const Minimal: Story<TrayProps> = (props) => (
  <ThemeProvider theme={getTheme()}>
    <ExomeTray {...props}>
      <div style={{ padding: 20 }}>Hello Tray</div>
    </ExomeTray>
  </ThemeProvider>
);

Minimal.args = {
  open: true,
  anchor: "left"
} as TrayProps;
