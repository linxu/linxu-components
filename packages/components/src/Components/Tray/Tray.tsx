import React from "react";
import { Drawer } from "@material-ui/core";

export interface TrayProps {
  // Side from which the Tray will appear anchor: top | bottom | left | right */
  anchor: "left" | "top" | "right" | "bottom";
  // The duration for the transition, in milliseconds */
  transitionDuration?: number;
  // The contents of the Tray */
  children: React.ReactElement;
  // If true, the Tray is open. */
  open: boolean;
  // Callback fired when the component requests to be closed */
  onClose: () => void;
}

// Tray are anchored to the top | bottom | left | right of the screen. <br>
// Other Props Reference https://material-ui.com/api/drawer/
export const Tray = (props: TrayProps) => {
  const { anchor, open, transitionDuration, onClose, children } = props;
  return (
    <Drawer
      anchor={anchor}
      open={open}
      onClose={onClose}
      transitionDuration={transitionDuration || 300}
    >
      {children}
    </Drawer>
  );
};
export const ExomeTray = Tray;
