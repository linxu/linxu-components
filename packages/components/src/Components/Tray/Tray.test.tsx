import React from "react";
import { render } from "@testing-library/react";
import { Tray } from "./Tray";

it("render Tray", async () => {
  const tree = render(
    <Tray open anchor="left" onClose={jest.fn()}>
      <h4>Test</h4>
    </Tray>
  );
  expect(tree).toMatchSnapshot();
});
