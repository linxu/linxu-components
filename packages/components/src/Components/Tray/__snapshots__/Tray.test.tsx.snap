// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Tray 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-modal"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiPaper-elevation16"
        tabindex="-1"
      >
        <h4>
          Test
        </h4>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
